decidim-core/app/controllers/decidim/qr_controller.rb

```diff
@@ -32,6 +31,12 @@ def current_participatory_space
 
     private
 
+    def validate_resource
```

Comment:
I think we should take into account the "matrioska" of unpublished related things: 

- return 404 when the component is unpublished
- return 404 when the space is unpublished


Can you add those checks and also specs for these cases?

Reply:
I have added those as 2 separate methods, as the `validate_resource` had a very big cyclomatic complexity.